packages/context/src/provider.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ValueOrPromise} from './binding';
+
+/**
+ * Provider of a value
+ */
```

Comment:
We probably should have more information here for the doc, including an example.